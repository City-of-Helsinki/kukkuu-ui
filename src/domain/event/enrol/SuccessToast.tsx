import React, { FunctionComponent, useState, useEffect } from 'react';
import { CSSTransition } from 'react-transition-group';
import { useTranslation } from 'react-i18next';
import { useDispatch, useSelector } from 'react-redux';

import styles from './successToast.module.scss';
import tada from '../../../assets/icons/svg/tada.svg';
import Icon from '../../../common/components/icon/Icon';
import { justEnrolled } from '../state/EventActions';
import { justEnrolledSelector } from '../state/EventSelectors';

const SuccessToast: FunctionComponent = () => {
  const { t } = useTranslation();
  const dispatch = useDispatch();

  const [showJustEnrolled, setShowJustEnrolled] = useState(false);
  const isJustEnrolled = useSelector(justEnrolledSelector);

  useEffect(() => {
    setShowJustEnrolled(isJustEnrolled);
  }, [isJustEnrolled]);

  let timer: NodeJS.Timeout;
  useEffect(() => {
    return () => {
      clearTimeout(timer);
    };
  });

  return (
    <CSSTransition
      className={styles.successToast}
      onClick={() => {
        setShowJustEnrolled(false);
      }}
      in={showJustEnrolled}
      timeout={{ appear: 300, enter: 300, exit: 300 }}
      classNames={{ ...styles }}
      onEntered={() => {
        timer = setTimeout(() => {
          setShowJustEnrolled(false);
        }, 3000);
      }}
      onExited={() => {
        clearTimeout(timer);
        dispatch(justEnrolled());
      }}
      unmountOnExit
    >
      <div
        onClick={() => {
          setShowJustEnrolled(false);
        }}
      >
        <Icon src={tada} className={styles.tadaIcon} />
        <div>
          <h1>{t('enrollment.successToast.heading')}</h1>
          <p>{t('enrollment.successToast.paragraph')}</p>
        </div>
      </div>
    </CSSTransition>
  );
};
export default SuccessToast;
