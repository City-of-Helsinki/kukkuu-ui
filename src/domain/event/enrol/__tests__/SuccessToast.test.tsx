import React from 'react';
import toJson from 'enzyme-to-json';

import SuccessToast from '../SuccessToast';
import { mountWithProvider } from '../../../../common/test/testUtils';

it('renders snapshot correctly', () => {
  const element = mountWithProvider(<SuccessToast />);
  expect(toJson(element)).toMatchSnapshot();
});
