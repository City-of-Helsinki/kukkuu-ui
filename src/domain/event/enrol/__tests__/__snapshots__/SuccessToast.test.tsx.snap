// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders snapshot correctly 1`] = `
<div>
  <div
    class="_successToast_663062"
  >
    <div
      class="_imgWrapper_73ff55 _tadaIcon_663062"
    >
      <img
        alt=""
        src="/icons/svg/tada.svg"
      />
    </div>
    <div>
      <h1>
        Tapahtumaan ilmoittautuminen onnistui!
      </h1>
      <p>
        Lähetimme varmistusviestin sähköpostiisi, löydät sieltä tapahtuman lisätiedot
      </p>
    </div>
  </div>
</div>
`;
